modules/spring-config/src/main/java/org/mule/config/spring/MuleArtifactContext.java

```diff
@@ -58,10 +79,35 @@ public MuleArtifactContext(MuleContext muleContext, Resource[] springResources)
     protected void prepareBeanFactory(ConfigurableListableBeanFactory beanFactory)
     {
         super.prepareBeanFactory(beanFactory);
-        beanFactory.addBeanPostProcessor(new MuleContextPostProcessor(muleContext));
-        beanFactory.addBeanPostProcessor(new ExpressionEvaluatorPostProcessor(muleContext));
-        beanFactory.addBeanPostProcessor(new GlobalNamePostProcessor());
-        beanFactory.registerSingleton(MuleProperties.OBJECT_MULE_CONTEXT, muleContext);
+
+        registerEditors(beanFactory);
+
+        addBeanPostProcessors(beanFactory,
+                              new MuleContextPostProcessor(muleContext),
+                              new ExpressionEvaluatorPostProcessor(muleContext),
+                              new GlobalNamePostProcessor(),
+                              new ExpressionEnricherPostProcessor(muleContext),
+                              new AnnotatedTransformerObjectPostProcessor(muleContext),
+                              new PostRegistrationActionsPostProcessor((MuleRegistryHelper) muleContext.getRegistry()),
```

Comment:
This downcast is ugly. PostRegistrationActinosPostProcessor should not depend on MuleRegistryHelper